frontend/src/components/ActivityField.tsx

```diff
@@ -1,15 +1,15 @@
-import { Show, createSignal } from "solid-js";
+import { Show, createEffect, createSignal } from "solid-js";
 
 import { LeftArrowIcon, RightArrowIcon } from "./UtilityIcons";
 
 function ActivityField(props) {
   const fieldNames = props.fields;
 
-  const currentIndex = fieldNames.indexOf(props.currentIndex);
+  const [field, setField] = createSignal(props.index - 1);
```

Comment:
We should check if the index is in the bounds of the field and set it to 0 if it is not.